repo.go

```diff
@@ -530,7 +738,11 @@ func (r *Repo) Sign(roleFilename string) error {
 		return err
 	}
 
-	keys, err := r.getSortedSigningKeys(role)
+	db, err := r.topLevelKeysDB()
```

Comment:
does this mean that `Sign` will only work for top-level roles? I would expect any delegation is allowed to be signed.

Reply:
This is easy to support and was an oversight. I added this in https://github.com/theupdateframework/go-tuf/pull/175/commits/189af3ec7bb2d58c4c0e5f3a8049a5a94ae72a66.

Reply:
I tried to write a unit test for `Sign` and `AddOrUpdateSignature` using only public methods, but couldn't find a way to create a new unsigned metadata file for a delegated role. I think we would need to make `Repo.setMeta` public to support that, and I'm not sure we want to do that in this same PR. Did you have code that needed to use `Sign` or `AddOrUpdateSignature`? If so, how were you creating the unsigned meta files?

Reply:
Yeah! I was able to create unsigned meta files. 

Shouldn't this function do it? https://github.com/theupdateframework/go-tuf/blob/0f60bf5964080d24bd5ad7c9847b87aaca6f4d9f/repo.go#L579

So Init repo, gen top-level keys, add targets and delegated role -- the delegated role addition doesn't include adding the keys to the store, so you'll end up with it unsigned. See example [here](https://github.com/asraa/test-sigstore-root/blob/765a319bd3778a6661e69fc9b816e535b2bf1a0c/cmd/tuf/app/add-delegation.go#L89-L112) (Note: i am using a slightly older API)

Reply:
In unit tests, previously what I've been doing is calling `localStore.SaveSigner` with the delegated role key before I call `repo.AddDelegatedRole`.

I added unit tests in https://github.com/theupdateframework/go-tuf/pull/175/commits/2d5f10dfd0ea3e5a6b766cc13739a93ffef6e1c8 that test `Sign` and `AddOrUpdateSignature` with the usage pattern you described.